MapboxNavigation/MapboxVoiceController.swift

```diff
@@ -28,15 +28,17 @@ open class MapboxVoiceController: RouteVoiceController, AVAudioPlayerDelegate {
     
     var audioTask: URLSessionDataTask?
     var cache: BimodalDataCache
+    let audioPlayerType: AVAudioPlayer.Type
     
     var speech: SpeechSynthesizer
     var locale: Locale?
     
     let localizedErrorMessage = NSLocalizedString("FAILED_INSTRUCTION", bundle: .mapboxNavigation, value: "Unable to read instruction aloud.", comment: "Error message when the SDK is unable to read a spoken instruction.")
 
-    @objc public init(speechClient: SpeechSynthesizer = SpeechSynthesizer(accessToken: nil), dataCache: BimodalDataCache = DataCache()) {
+    @objc public init(speechClient: SpeechSynthesizer = SpeechSynthesizer(accessToken: nil), dataCache: BimodalDataCache = DataCache(), audioPlayerType: AVAudioPlayer.Type? = nil) {
         speech = speechClient
         cache = dataCache
+        self.audioPlayerType = audioPlayerType ?? AVAudioPlayer.self
```

Comment:
Should this default be declared in the initializer instead?

Reply:
I generally like to take this approach to default-arguments for two reasons:

1. It prevents you from having to statically determine "will I or won't I override?" at the call-site. It's much easier to decide to pass nil or not at runtime then it is to figure out overridability logic at compile-time. Basically if you expose the argument in the call, but you still want the default logic, you have to recreate it. Not so with optional defaults.
2. It's way, _way_ more ergonomic for an OBJ-C dev to simply pass "nil" if they want default logic than it is for them to recreate default behavior, and it prevents us from having to make OBJ-C wrapper functions/inits all over the place.

Reply:
Good points. Should we make this pattern more consistent? Or do the existing args pass the smell test?

It's probably beyond the scope of this PR though